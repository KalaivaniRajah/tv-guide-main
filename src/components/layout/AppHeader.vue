<template>
  <div class="container-fluid">
    <nav class="navbar navbar-expand-md navbar-light bg-light">
      <h1><router-link to="/">TV Guide</router-link></h1>
      <button
        type="button"
        class="navbar-toggler"
        data-toggle="collapse"
        data-target="#navbarCollapse"
      >
        <span class="navbar-toggler-icon"></span>
      </button>
      <div class="collapse navbar-collapse" id="navbarCollapse">
        <ul class="form-inline ml-auto">
          <div class="container">
            <div class="row input-row">
              <input
                id="userInput"
                class="form-control mr-sm-2"
                type="search"
                placeholder="Search"
                aria-label="Search"
                v-model="inputField"
              />
            </div>
            <div class="row" style="padding: 0px; margin: 0px; height: 0px">
              <div
                class="invalid-feedback"
                style="visibility: hidden; margin-top: 0px; height: 5px"
              >
                Enter a valid Show name!
              </div>
            </div>
          </div>
          <li>
            <button-layout id="searchButton" @click="findShow"
              >Search
            </button-layout>
          </li>
        </ul>
      </div>
    </nav>
  </div>
</template>

<script>
import { mapGetters } from "vuex";
import ButtonLayout from "./ButtonLayout.vue";
export default {
  components: { ButtonLayout },
  data() {
    return {
      searchedShow: null,
      inputField: "",
      allShows: [],
    };
  },
  computed: {
    ...mapGetters({
      topShows: "shows/shows",
    }),
  },
  methods: {
    findShow() {
      this.allShows = this.topShows;
      if (this.inputField) {
        this.searchedShow = this.allShows.find((show) => {
          return show.name.toLowerCase() === this.inputField.toLowerCase();
        });
        if (this.searchedShow) {
          document.getElementsByClassName(
            "invalid-feedback"
          )[0].style.visibility = "hidden";
          this.$router.push("/showdetails/" + this.searchedShow.id);
        } else {
          document.getElementsByClassName(
            "invalid-feedback"
          )[0].style.visibility = "visible";
        }
      } else {
        document.getElementsByClassName(
          "invalid-feedback"
        )[0].style.visibility = "visible";
      }
    },
  },
};
</script>

<style scoped>
.container {
  width: 100%;
  padding-right: 0px;
  padding-left: 0px;
  margin-right: auto;
  margin-left: 0px;
}
div {
  width: 100%;
  height: 5rem;
  display: flex;
  justify-content: center;
  align-items: center;
}

div a {
  text-decoration: none;
  color: #3a0061;
  display: inline-block;
  border: 1px solid transparent;
}

h1 a:hover,
h1 a:active,
h1 a.router-link-active {
  border-color: transparent;
}

div nav {
  width: 90%;
  margin: auto;
  display: contents;
  justify-content: space-between;
  align-items: center;
}

div ul {
  list-style: none;
  margin: 0;
  padding: 0;
  display: contents;
  justify-content: center;
  align-items: center;
}

li,
input {
  margin: 0 0.5rem;
}

.input-row {
  margin: 10px;
  height: 20px;
}
</style>
