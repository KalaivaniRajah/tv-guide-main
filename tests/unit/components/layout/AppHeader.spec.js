import { shallowMount } from "@vue/test-utils";
import AppHeader from "../../../../src/components/layout/AppHeader.vue";
import { createStore } from "vuex";

describe("AppHeader", () => {
  test("by default the input field should be empty and the searchedShow should be null", () => {
    const wrapper = shallowMount(AppHeader);
    expect(wrapper.vm.$data.searchedShow).toEqual(null);
  });

  test("if we click the Search button, it should call the findShow method and the searchedShow value should get changed", () => {
    const mockMethod = jest.spyOn(AppHeader.methods, 'findShow');
    const wrapper = shallowMount(AppHeader, {
    });
    wrapper.vm.store = createStore({
      modules: {
        shows: {
          namespaced: true,
          state() {
            return {
              shows: [
              {
                id: 83,
                name: "The Simpsons",
                image: "https://static.tvmaze.com/uploads/images/medium_portrait/271/679902.jpg",
                network: "FOX",
                genre: "Comedy",
                summary: "<p><b>The Simpsons</b> is the longest running scripted show in US television history. It captures the adventures of Homer, Marge, Maggie, Bart and Lisa who are living in a fictional town called Springfield.</p>",
                status: "Running",
                rating: 8.2,
                premiered: "1989-12-17",
                site: "http://www.fox.com/the-simpsons/full-episodes",
              }
              ]
            }
          },
          getters: {
            shows(state) {
              return state.shows;
            },
          }
        }
      }
    });
    const mockRouter = {
      push: jest.fn()
    }
    wrapper.find("#userInput").setValue("The Simpsons");
    wrapper.vm.$store = wrapper.vm.store;
    wrapper.vm.$router = mockRouter;
    wrapper.find("#searchButton").trigger("click");
    expect(mockMethod).toHaveBeenCalled();
    expect(mockRouter.push).toHaveBeenCalledTimes(1);
    expect(mockRouter.push).toHaveBeenCalledWith("/showdetails/83");
  });
});