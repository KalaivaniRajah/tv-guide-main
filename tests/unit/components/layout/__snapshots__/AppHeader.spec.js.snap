// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader should render to a snapshot when list is empty 1`] = `
<div
  class="container-fluid"
>
  <nav
    class="navbar navbar-expand-md navbar-light bg-light"
  >
    <h1>
      <router-link
        to="/"
      >
        TV Guide
      </router-link>
    </h1>
    <button
      class="navbar-toggler"
      data-target="#navbarCollapse"
      data-toggle="collapse"
      type="button"
    >
      <span
        class="navbar-toggler-icon"
      />
    </button>
    <div
      class="collapse navbar-collapse"
      id="navbarCollapse"
    >
      <ul
        class="form-inline ml-auto"
      >
        <input
          aria-label="Search"
          class="form-control mr-sm-2"
          id="userInput"
          placeholder="Search"
          type="search"
        />
        <li>
          <button-layout-stub />
        </li>
      </ul>
    </div>
  </nav>
</div>
`;
